scripts/world_bank/worldbank.py

```diff
@@ -0,0 +1,393 @@
+variableMeasured: C:WorldBank->StatisticalVariable
+observationDate: C:WorldBank->Year
+observationPeriod: "P1Y"
+observationAbout: C:WorldBank->IsoCode
+value: C:WorldBank->Value
+"""
+
+TEMPLATE_STAT_VAR = """
+Node: dcid:WorldBank/{INDICATOR}
+name: "{NAME}"
+description: "{DESCRIPTION}"
+typeOf: dcs:StatisticalVariable
+populationType: dcs:{populationType}
+statType: dcs:{statType}
+measuredProperty: dcs:{measuredProperty}
+measurementDenominator: dcs:{measurementDenominator}
+{CONSTRAINTS}
+"""
+
+COUNTRY_MCF = """
+Node: E:WorldBank->E1
+typeOf: dcs:Country
+countryAlpha3Code: C:WorldBank->IsoCode
+"""
+
+def read_worldbank(country_iso):
+    """ Fetches and tidies all ~1500 Worldbank indicators
+        for a given 3 digit ISO code.
+
+        For a particular 3 digit ISO code, this function fetches the entire ZIP
+        file for that particular country for all Worldbank indicators.
+        The format of the file has rows for each economic indicator for
+        each year. The data is tidied so that each year is a column.
+
+        Args:
+            country_iso: 3-digit ISO code for a country, as a string.
+
+        Returns:
+            A tidied pandas dataframe with all indicator codes for a particular
+            country in the format of (country, indicator, year, value).
+
+        Notes:
+            Takes approximately 10 seconds to download and
+            tidy one country in Google colab.
+    """
+    country_zip = ("http://api.worldbank.org/v2/en/country/"
+        + country_iso + "?downloadformat=csv")
+    r = requests.get(country_zip)
+    filebytes = io.BytesIO(r.content)
+    myzipfile = zipfile.ZipFile(filebytes)
+
+    # We need to select the data file which starts with "API",
+    # but does not have an otherwise regular structure.
+    file_to_open = None
+    for file in myzipfile.namelist():
+        if file.startswith("API"):
+            file_to_open = file
+            break
+    assert file_to_open is not None
+
+    df = None
+    # Captures any text contained in double quotatations.
+    line_match = re.compile(r"\"([^\"]*)\"")
+
+    for line in myzipfile.open(file_to_open).readlines():
+        # Cells are contained in quotations and comma separated.
+        cols = line_match.findall(line.decode("utf-8"))
+
+        # CSVs include header informational lines which should be ignored.
+        if len(cols) > 2:
+            # Use first row as the header.
+            if df is None:
+                df = pd.DataFrame(columns=cols)
+            else:
+                df = df.append(pd.DataFrame([cols], columns=df.columns),
+                               ignore_index=True)
+
+    df = df.rename(columns=WORLDBANK_COL_REMAP)
+
+    # Turn each year into its own row.
+    df = df.set_index(
+        ['CountryName', 'CountryCode', 'IndicatorName', 'IndicatorCode'])
+    df = df.stack()
+    df.index = df.index.rename('year', level=4)
+    df.name = "Value"
+    df = df.reset_index()
```

Comment:
Does Pandas's wide_to_long or melt suffice?

https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.wide_to_long.html
https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.melt.html

Reply:
I believe you would have to create a list of all years to use melt.